packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaders.tsx

```diff
@@ -26,13 +26,11 @@ const GridColumnHeadersRoot = styled('div', {
   overridesResolver: (props, styles) => styles.columnHeaders,
 })(({ theme }) => {
   return {
-    position: 'absolute',
-    top: 0,
-    left: 0,
-    right: 0,
+    position: 'relative',
     overflow: 'hidden',
     display: 'flex',
     alignItems: 'center',
+    boxSizing: 'border-box',
```

Comment:
This is a bugfix actually, you can see that in v5 the column header border is not visible when you have pinned rows:
https://codesandbox.io/s/strange-sea-6ul544?file=/demo.tsx
You cannot reproduce this in [the same demo in the docs](https://mui.com/x/react-data-grid/row-pinning/) because in the docs we use CssBaseline which sets `* { box-sizing: border-box }`.

I didn't want to extract this into a separate PR to save time.
I'll cherry-pick this change to master once this PR is merged.